AuthenticationDemo/ExampleSessionService.swift

```diff
@@ -125,3 +141,65 @@ public final class ExampleSessionService: SessionServiceType {
     }
     
 }
+
+fileprivate extension ExampleSessionService {
+
+    fileprivate func bindPersistance() {
```

Comment:
This is open to the Session Service you want to implement. Most surely you would want persistance so that the app is friendly to the user, but maybe you need to fetch the user with API before starting the AuthenticationCoordinator and you don't use local persistance. This is just one example of a Session Service so the user can imagine how they can solve it. And to see how to handle it with providers.